src/GraphQL.Tests/Execution/ResolveFieldContextTests.cs

```diff
@@ -199,6 +199,33 @@ public async Task try_resolve_async_properly_resolves_result()
             result.ShouldBe("Test Result");
         }
 
+        [Fact]
+        public void resolveFieldContextAdapter_throws_error_when_null()
+        {
+            Should.Throw<ArgumentNullException>(() =>
+            {
+                var adapter = new ResolveFieldContextAdapter<object>(null);
+            });
+        }
+
+        [Fact]
+        public void resolveFieldContextAdapter_throws_error_if_invalid_type()
+        {
+            var context = new ResolveFieldContext() { Source = "test" };
+            Should.Throw<ArgumentException>(() =>
+            {
+                var adapter = new ResolveFieldContextAdapter<int>(context);
+            });
+        }
+
+        [Fact]
+        public void resolveFieldContextAdapter_accepts_null_sources()
+        {
+            var context = new ResolveFieldContext();
+            var adapter = new ResolveFieldContextAdapter<int>(context);
+            adapter.Source.ShouldBe(0);
+        }
```

Comment:
I think that behavior should be different for value and ref+nullable types.

```c#
 [Fact]
        public void resolveFieldContextAdapter_accepts_null_sources_for_reference_types()
        {
            var context = new ResolveFieldContext();
            var adapter = new ResolveFieldContextAdapter<string>(context);
            adapter.Source.ShouldBeNull();
        }
 [Fact]
        public void resolveFieldContextAdapter_accepts_null_sources_for_nullable_types()
        {
            var context = new ResolveFieldContext();
            var adapter = new ResolveFieldContextAdapter<int?>(context);
            adapter.Source.ShouldBeNull();
        }
 [Fact]
        public void resolveFieldContextAdapter_throw_on_null_sources_for_value_types()
        {
            var context = new ResolveFieldContext();
            Should.Throw<ArgumentException>(() => new ResolveFieldContextAdapter<int>(context));
        }
```

Reply:
I'm open to suggestions as to how best to accomplish this.

Reply:
With runtime check inside ResolveFieldContextAdapter.cs